flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -519,7 +519,7 @@ JobManagerRunner createJobManagerRunner(JobGraph jobGraph, long initializationTi
 
                     return Acknowledge.get();
                 },
-                jobManagerSharedServices.getScheduledExecutorService());
+                jobManagerSharedServices.getFutureExecutor());
```

Comment:
Even though it is still an asynchronous completable future? Just to make sure I follow, since it is performing an IO action even though it is not blocking we should use the IOExecutor?

Reply:
I think that `Checkpoints.disposeSavepoint` can read from disk (also remote) and this can be blocking. Hence, my suggestion.